src/main/java/org/ossgang/commons/observables/operators/connectors/AbstractConnectorObservableValue.java

```diff
@@ -49,6 +51,7 @@ protected void connect(ObservableValue<T> upstream) {
                     (self, exception) -> self.dispatchException(exception),
                     AbstractConnectorObservableValue::subscriberCountChanged), FIRST_UPDATE);
             connectionState.set(CONNECTED);
+            ongoingConnection = false;
```

Comment:
Consider try...finally, otherwise the flag will not be reset if requireNonNull or subscribe throws